flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -359,6 +366,16 @@ private void setChaining(Map<Integer, byte[]> hashes, List<Map<Integer, byte[]>>
         }
     }
 
+    private static int compareHashes(byte[] hash1, byte[] hash2) {
+        for (int index = 0; index < hash1.length; index++) {
+            int diff = hash2[index] - hash1[index];
```

Comment:
nit: handle arrays of different length

Reply:
Hashes are currently always of the same size. However, there is no contract on the `StreamGraphHasher` that enforces that, so I'll incorporate your idea.